py/tests/test_fetcher.py

```diff
@@ -0,0 +1,207 @@
+                self.good_response.get("series"),
+            )
+        )
+        assert parsed_unit == "bytes"
+
+    def test_parse_and_assert_unit_no_unit(self) -> None:
+        series = self.good_response["series"]
+        series[0].pop("unit")  # type: ignore
+        self.assertRaises(
+            AssertionError,
+            ddf.parse_and_assert_unit,
+            cast(ddf.DatadogResponseSeries, series),
+        )
+
+    def test_warn_multiple_units(self) -> None:
+        series_list = self.good_response["series"]
+        series_list[0]["unit"][1] = {"dummy": "unit"}  # type: ignore
+        with self.assertLogs("fetcher") as cm:
+            ddf.warn_multiple_units(
+                cast(Sequence[ddf.DatadogResponseSeries], series_list)
+            )
+            assert len(cm.output) == 1
+
+    def test_parse_and_assert_response_series(self) -> None:
+        series_list = ddf.parse_and_assert_response_series(self.good_response)
+        assert len(series_list) == 1
+
+    def test_parse_and_assert_response_series_error(self) -> None:
+        self.assertRaises(
+            AssertionError,
+            ddf.parse_and_assert_response_series,
+            self.bad_response,
+        )
+
+    def test_parse_and_assert_response_scope(self) -> None:
+        sample_scope = self.good_response["series"][0].get(  # type: ignore
+            "scope"
+        )
+
+        self.assertDictEqual(
+            ddf.parse_and_assert_response_scope(sample_scope),
+            cast(
+                Mapping[str, str],
+                self.processed_response["series"][0]["scope_dict"],
+            ),
+        )
+
+    def test_parse_and_assert_response_scope_no_app_feature(self) -> None:
+        self.assertRaises(
+            AssertionError,
+            ddf.parse_and_assert_response_scope,
+            '"shared_resource_id": "rc_long_redis"',
+        )
+
+    def test_process_series_data(self) -> None:
+        expected_record_list = [
+            ddf.UsageAccumulatorRecord(
+                resource_id="rc_long_redis",
+                app_feature="shared",
+                amount=2,
+                usage_type=accumulator.UsageUnit.BYTES,
+            ),
+            ddf.UsageAccumulatorRecord(
+                resource_id="rc_long_redis",
+                app_feature="shared",
+                amount=3,
+                usage_type=accumulator.UsageUnit.BYTES,
+            ),
+        ]
+
+        self.assertEqual(
+            ddf.process_series_data(
+                cast(
+                    Sequence[ddf.DatadogResponseSeries],
+                    self.processed_response["series"],
+                ),
+                accumulator.UsageUnit.BYTES,
+            ),
+            expected_record_list,
+        )
+
+    @patch("usageaccountant.datadog_fetcher.query_datadog")
+    @patch("usageaccountant.accumulator.UsageAccumulator.__init__")
+    @patch("usageaccountant.accumulator.UsageAccumulator.record")
```

Comment:
Please do not patch these. By patching code you can safely run you are reducing your own test coverage for no reason. 
If you want to mock flush, that's ok, but please do not mock the other methods.